product.py

```diff
@@ -566,6 +580,22 @@ def findXenSourceBackups():
             backup = XenServerBackup(p, b.mount_point)
             logger.log("Found a backup: %s" % (repr(backup),))
 
+            # Don't restore a BIOS backup with a UEFI installer and conversely
+            backup_is_uefi = False
+            with open(os.path.join(b.mount_point, 'etc', 'fstab'), 'r') as fstab:
+                for line in fstab:
+                    m = re.search("/boot/efi[[:space:]]+vfat", line)
```

Comment:
do we really need to check the fstype?
I suggest something like `if line.split(maxsplit=1)[0] == "/boot/efi"`

Reply:
Actually changed it to `re.search(r'^[^#].*\s+/boot/efi', line.strip())` to ignore commented lines

Reply:
still, a regex seems overkill: we only need to check for the first word, and `split()` will even take care of any leading whitespace

Reply:
As discussed, we'll stick with a `re.search()`